### PR TITLE
Test result submission displays delivery failure modal on invalid phone numbers

### PR DESCRIPTION
**Pull request by nathancrtr**
## Related Issue or Background Info

#1651 

## Changes Proposed

- Backend
    - Add `AddTestResultResponse` API model class to add additional context to a `TestOrder`
        - `addTestResult` mutation resolver returns an instance of this class
    -  Add corresponding `AddTestResultResponse` type in GraphQL schema
- Frontend
    - Patient phone number appears on AoE form when electing to send test results by SMS
    - If patient has requested to receive test results via SMS but has provided an invalid phone number, a Toast alert will appear upon test result submission indicating that the delivery was not successful

## Additional Information
* As a result of a Slack conversation with @hmyers-cms (https://skylight-hq.slack.com/archives/C01SY3DRMNY/p1623260681151300), the requirements for this ticket were changed from the original ticket description.
    * We have opted to display the "delivery failure" message to the user as a Toast alert instead of a modal

## Screenshots / Demos
* Branch deployed to `test` environment to test Twilio integration:

https://user-images.githubusercontent.com/27730981/121723081-1595e080-cab4-11eb-81d9-a88cd01fcdbc.mov

## Checklist for Author and Reviewer
- [x] @hmyers-cms: ~does the appearance of the modal look OK to you? The design image in the ticket just had the modal out of context of the page it appears on. Are we picky about how it appears on mobile?~
    - See follow-up in `Additional Information` section

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] ~Database changes are submitted as a separate PR~
  - [x] ~Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user~
  - [x] ~Any changes to tables that have custom no-PHI views are accompanied by changes to those views~
- [x] ~GraphQL schema changes are backward compatible with older version of the front-end~

### Security
- [x] ~Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)~
- [x] ~Any dependencies introduced have been vetted and discussed~

## Cloud
- [x] ~DevOps team has been notified if PR requires ops support~
- [x] ~If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification~

### CODE REVIEW

**Review by nullflux — CHANGES_REQUESTED** [review]
LGTM except for the bit about it being backwards compatible? Let's verify that before moving ahead

Also pending UX feedback from @hmyers-cms

File: backend/src/main/resources/graphql/main.graphqls
Comment (nullflux) [reply backend/src/main/resources/graphql/main.graphqls:408]: ☝️ Resolved in pairing

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/AddTestResultResponse.java
Comment (nathancrtr) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/AddTestResultResponse.java]: Done in 3fad430.

File: frontend/src/app/testQueue/QueueItem.tsx
Comment (nathancrtr) [reply frontend/src/app/testQueue/QueueItem.tsx]: Not necessary. Removed in 3fad430.

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java
Comment (nathancrtr) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java:163]: I think this was an unruly format-on-save. I've disabled that in VSCode for now.

**Comment by nathancrtr** [comment]
@nullflux All issues have been addressed and all checks are passing. Please review when you are able and approve if your requested changes have been resolved.

File: frontend/src/app/testQueue/QueueItem.tsx
Comment (nullflux) [inline frontend/src/app/testQueue/QueueItem.tsx:432]: Does this change belong on this branch? I feel like it's from the refactor in #1820 that DRYs up this formatting into utils

Otherwise, LGTM ✅ but please double check this before merging

**Review by nullflux — APPROVED** [review]
Provisionally approved - please see my new comment before merging

File: frontend/src/app/testQueue/QueueItem.tsx
Comment (nathancrtr) [reply frontend/src/app/testQueue/QueueItem.tsx]: Addressed in a39cfc3.

**Pull request merged**
